crashsimilarity/models/doc2vec.py

```diff
@@ -0,0 +1,53 @@
+import multiprocessing
+import os
+import random
+import time
+import logging
+
+import gensim
+from datetime import datetime
+
+from crashsimilarity import utils
+from crashsimilarity.models.base import EmbeddingAlgo
+
+
+class Doc2Vec(EmbeddingAlgo):
+    def _read_corpus(self):
+        return [gensim.models.doc2vec.TaggedDocument(trace, [i, signature]) for i, (trace, signature) in enumerate(self._read_traces())]
+
+    def _extract_words_from_model(self, doc_id):
+        return [w for w in self._corpus[doc_id].words if w in self._model]
+
+    def _extract_indices_from_model(self, doc_id):
+        return [self._model.wv.vocab[word].index for word in self._corpus[doc_id].words if word in self._model]
+
+    def _train_model(self, force_train=False):
```

Comment:
There are a lot of similarities between the two `ŧrain_model` functions in Word2Vec and Doc2Vec, but we can refactor in a follow-up.